Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -453,21 +454,21 @@ private void NetworkManager_NetworkChanged(object sender, EventArgs e)
         /// Gets the current application version
         /// </summary>
         /// <value>The application version.</value>
-        public Version ApplicationVersion => _version ?? (_version = typeof(ApplicationHost).Assembly.GetName().Version);
+        public Version ApplicationVersion => _version ?? (_version = ApplicationExtendedVersion.ApiVersion);
 
-        private string _serverVersion;
+        private Version _serverVersion;
         /// <summary>
         /// Gets the current application server version
         /// </summary>
         /// <value>The application server version.</value>
-        public string ApplicationServerVersion => _serverVersion ?? (_serverVersion = FileVersionInfo.GetVersionInfo(Assembly.GetEntryAssembly().Location).FileVersion);
+        public Version ApplicationServerVersion => _serverVersion ?? (_serverVersion = typeof(ApplicationHost).Assembly.GetName().Version);
 
-        private BuildVersion _buildVersion;
+        private ExtendedVersion _extendedVersion;
         /// <summary>
         /// Gets the current application server version
         /// </summary>
         /// <value>The application server version.</value>
-        public BuildVersion ApplicationBuildVersion => _buildVersion ?? (_buildVersion = BuildVersion.FromSimpleFile("jellyfin.version"));
+        public ExtendedVersion ApplicationExtendedVersion => _extendedVersion ?? (_extendedVersion = typeof(ApplicationHost).Assembly.GetCustomAttributes(typeof(AssemblyExtendedVersion), false).Cast<AssemblyExtendedVersion>().FirstOrDefault().ExtendedVersion);
```

Comment:
Can this long line be wrapped? Hard to read the diff on github...

Reply:
Done.